app/com/linkedin/drelephant/tunin/JobCompleteDetector.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.drelephant.tunin;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import models.JobExecution;
+import models.TuningJobExecution;
+import models.TuningJobExecution.ParamSetStatus;
+
+import org.apache.log4j.Logger;
+
+import play.libs.Json;
+
+
+/**
+ * This class pools the scheduler for completion status of execution and updates the database with current status
+ * of the job.
+ */
+public abstract class JobCompleteDetector {
```

Comment:
Why is this a separate class? Can we package all the scheduler specific functionalities together under one module?

Reply:
As discussed this is specific to auto tuning, so it's part of tunin package.